tests/suites/test_suite_ssl.function

```diff
@@ -1755,3 +1981,76 @@ void ssl_session_serialize_version_check( int corrupt_major,
 
 }
 /* END_CASE */
+
+/* BEGIN_DEPENDENCIES
+ * depends_on:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_PEM_PARSE_C:MBEDTLS_RSA_C:MBEDTLS_ECDSA_C
+ * END_DEPENDENCIES
+ */
+
+/* BEGIN_CASE */
+void test_handshake_state()
+{
+    enum { BUFFSIZE = 1024 };
+    int ret = 0;
+    mbedtls_endpoint ep;
+    mbedtls_endpoint_certificate cert;
+    mbedtls_mock_socket socket; /* dummy socket */
+
+    /* Test basic server initializing */
+
+    TEST_ASSERT( ( ret = mbedtls_endpoint_init( &ep, MBEDTLS_SSL_IS_SERVER ) )
+                 == 0 );
+    TEST_ASSERT( ( ret = mbedtls_endpoint_certificate_init( &ep, &cert ) )
+                 == 0 );
+    TEST_ASSERT( mbedtls_mock_socket_connect( &(ep.socket), &socket, BUFFSIZE )
+                 == 0 );
+
+    /* Test diffrent state of handshake for server */
+
+    TEST_ASSERT( ep.ssl.state == MBEDTLS_SSL_HELLO_REQUEST );
+    TEST_ASSERT( mbedtls_ssl_handshake_step( &(ep.ssl) ) == 0 );
+    TEST_ASSERT( ep.ssl.state == MBEDTLS_SSL_CLIENT_HELLO );
+
+    TEST_ASSERT( mbedtls_endpoint_reinit( &ep, MBEDTLS_SSL_IS_SERVER ) == 0 );
+    TEST_ASSERT( mbedtls_mock_socket_connect( &(ep.socket), &socket, BUFFSIZE ) == 0 );
+    ep.ssl.state = MBEDTLS_SSL_SERVER_NEW_SESSION_TICKET;
+    TEST_ASSERT( mbedtls_ssl_handshake_step( &(ep.ssl) ) != 0 );
```

Comment:
After we've successfully stepped after setting `state` to `MBEDTLS_SSL_SERVER_NEW_SESSION_TICKET`, are there any further test assertions we'd like to make or is only checking the return value good enough?

Reply:
We could replace '!= 0' into specific error which is expected. 
The truth is that setting the ssl state in such a way should be prohibited if properly used mbedtls library. After doing something like this the sll's structures are damaged and only what we can test is if mbedtls_ssl_handshake_step does not return success

Reply:
I am not sure what we are trying to test after doing the step. Are there any valuable changes to any structs we should check to ensure step worked correctly? Maybe just checking `!= 0` is good enough; I am not sure.